06_puppy/mohitnag/store_test.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storer struct {
+	suite.Suite
+	store      Storer
+	storerType func() Storer
+}
+
+func (s *storer) SetupTest() {
+	s.store = s.storerType()
+}
+
+func (s *storer) TestCreatePuppy() {
+	assert := assert.New(s.T())
+	puppy := Puppy{ID: 1, Breed: "dog", Colour: "white", Value: "$2"}
+	testCases := []struct {
+		Scenario      string
+		Input         Puppy
+		ExpectedError string
+	}{
+		{"Create Puppy", puppy, "<nil>"},
+		{"Creating already existing Puppy should fail", puppy, "puppy with Id 1 already exists"},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		s.T().Run(tc.Scenario, func(t *testing.T) {
+			err := s.store.CreatePuppy(tc.Input)
+			assert.Equal(tc.ExpectedError, fmt.Sprint(err))
+
+		})
+	}
+}
+
+func (s *storer) TestReadPuppy() {
+	assert := assert.New(s.T())
+	puppy := Puppy{ID: 1, Breed: "dog", Colour: "white", Value: "$2"}
```

Comment:
I think the puppy objects that u create within each test function could be created once and reused...

Reply:
Done. creating puppy object in test setup.